packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -607,14 +792,23 @@ export function mergeRequirements(
   rs1: GPCRequirements,
   rs2: GPCRequirements
 ): GPCRequirements {
+  // Either `rs1` or `rs2` specifies the tuple arities.
+  const tupleArities =
+    Object.keys(rs1.tupleArities).length === 0
+      ? rs2.tupleArities
+      : rs1.tupleArities;
   return GPCRequirements(
     Math.max(rs1.nObjects, rs2.nObjects),
     Math.max(rs1.nEntries, rs2.nEntries),
     Math.max(rs1.merkleMaxDepth, rs2.merkleMaxDepth),
-    rs1.nListElements.map((nElements, i) =>
-      Math.max(nElements, rs2.nListElements[i])
+    Math.max(rs1.nLists, rs2.nLists),
+    Object.fromEntries(
```

Comment:
This merge doesn't seem general enough for the case where the name sets don't match exactly, which is checked, but not before this merge happens.  Maybe there should be a check here, and/or maybe the merge in the caller should be moved to happen after all checks.